ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+
+#### Turns
+
+  * In Orleans a Grain is scheduled in terms of non-overlapping *turns*, meaning that no two threads run that Grain’s code concurrently at any given time. This is also true for continuations formed via the usage of async–await.
+  
+  * Akka schedules Actors similarly, upon each assignment to a (single) Thread an Actor may process up to a configurable limit of messages and it is guaranteed that an Actor is running on at most one Thread at any given time. While it is possible to specify an ExecutionContext that runs continuations within the Actor’s schedule, the default executors do not respect this constraint and will typically run Future continuations concurrently with the Actor that scheduled them.
+  
+#### Persistence
+
+  * Orleans provides persistence based on snapshots that are managed by the Grain explicitly. This maps well to a database-backed model where each Grain is responsible for a conceptual “row”.
+  
+  * Akka Persistence is based upon event-sourcing, which means that the PersistentActor emits events that represent state changes to be applied. Snapshots may be used as an optimization to shorten recovery time.
+
+In both cases the persistence provider is configurable and users may choose to persist Actor state independently by other means as well. The biggest difference between snapshots and event-sourcing is that the event history of a PersistentActor contains valuable information that is lost when persisting only snapshots: each event has a well-defined meaning in the business domain and thus the event log retains its meaning as the internal representation of the Actor’s state changes over time while snapshots will become invalid and cannot be mined for additional insights later.
+
+#### Timers and Reminders
+
+  * Orleans provides transient timers that are local to a Grain activation and cease to exist when the Grain is passivated. It also provides reliable persistent reminders that will send an invocation to a Grain independently of whether it is currently active or not.
+  
+  * Akka includes only transient timers at this point, reliable reminders need to be implemented using third-party extensions like Akka Quartz.
+
+### Runtime Implementation
+
+#### Overview
+
+Both systems consist of similar core components: Orleans’ Execution corresponds to akka-actor in that it is responsible for the actual execution of Actors, Messaging corresponds to akka-remote in that it manages TCP connections between the cluster nodes, and Hosting corresponds to akka-cluster plus ClusterSharding. One difference is that akka-actor also allows purely local message sends that are optimized to local reference passing whereas Orleans invocations are always sent via the Messaging component that consults with the Hosting component as to where the recipient currently resides. Akka’s lack of indirection allows substantial performance optimizations in this regard.
```

Comment:
I think you may have misunderstood one aspect: when sending a message to another Actor, that invocation is always asynchronous, the message always goes through the recipient’s mailbox (which in fact is the task that is scheduled on the thread pool). The only optimization that Akka does is to allow the user to send messages by reference without intermediate serialization—we do have a configuration option to turn on full serialization also in the local case in order to verify that your program does not in fact rely upon shared state. This means that we very much uphold the Actor Model, we just chose to have the nanny be opt-in while Orleans chose opt-out ;-)